Filter Holds

AlexanderBlanchardAC opened the pull request:
https://aspen-discovery.atlassian.net/jira/software/c/projects/DIS/boards/2?search=219&selectedIssue=DIS-219

This Pull Request adds filtering options to the holds section in a user account.

AlexanderBlanchardAC committed:
feat: Add Filtering for Holds

Add the ability to filter holds by excluding those of selected linked
users and by displaying only selected holds in the UI.

AlexanderBlanchardAC committed:
feat: Export Filtered Holds

Update exportHolds so that the new filters are taken into account when
holds are exported.

AlexanderBlanchardAC committed:
chore: Update change log

Chloe070196 commented:
Seems to be working nicely and as expected, would add return types to remove the need for them to be manually added upon merge with main

Edit: two minor issues found in further testing:

- Selecting a check box, then selecting a user to hide, and then selecting 'Display Selected' will have the same effect as selecting a user to hide, then selecting 'Hide Users'.
- If holds are placed on one same record by multiple linked users (and/or the active user), selecting one, then selecting 'Display Selected' will show all holds for that record, instead of only displaying the one.

AlexanderBlanchardAC committed:
refactor: Add return types

AlexanderBlanchardAC committed:
fix: Fix display selected filter filtering users

Prevent disply selected filter from filtering out linked users if they
are highlighted in the ui

AlexanderBlanchardAC committed:
fix: Fix returning all records with record id

When records are selected and display selected is applied, records were
being returned if they had the same record id as a selected record e.g.
they had a different user, but were not selected themseleves, this patch
adds a check for the userid to prevent this

Chloe070196 commented:
Looks to be working as expected with both issues noted during testing now resolved
